src/Chapter2.hs

```diff
@@ -841,7 +867,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n lst
+    | n < 0 = []
+    | otherwise = take (length lst) $ drop n $ cycle lst
```

Comment:
That's a correct solution! Though, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face: